app/Harvest/Importers/AbstractImporter.php

```diff
@@ -109,19 +110,21 @@ protected function processHasMany(Model $model, $field, array $relatedRows) {
             $updateIds[] = $instance->getKey();
         }
 
-        $relatedKeyName = $relation->getRelated()->getKeyName();
-        $relation->whereNotIn($relatedKeyName, $updateIds)->each(function (Model $related) {
-            $related->delete();
-        });
+        if ($allowDelete) {
+            $relatedKeyName = $relation->getRelated()->getKeyName();
+            $relation->whereNotIn($relatedKeyName, $updateIds)->each(function (Model $related) {
+                $related->delete();
+            });
```

Comment:
Would this also work?
```suggestion
            $relation->whereNotIn($relatedKeyName, $updateIds)->delete();
```

Reply:
The case of bulk delete model events wouldn't be fired
https://laravel.com/docs/7.x/eloquent#deleting-models